src/Trigger/RabbitMQListener.cs

```diff
@@ -140,12 +154,124 @@ internal void RepublishMessages(BasicDeliverEventArgs ea)
             }
         }
 
+        public string Id
+        {
+            get
+            {
+                return $"{_functionId}-RabbitMQTrigger-{_queueName}".ToLower();
+            }
+        }
+
         private void ThrowIfDisposed()
         {
             if (_disposed)
             {
                 throw new ObjectDisposedException(null);
             }
         }
+
+        async Task<ScaleMetrics> IScaleMonitor.GetMetricsAsync()
+        {
+            return await GetMetricsAsync();
+        }
+
+        public async Task<RabbitMQTriggerMetrics> GetMetricsAsync()

```

Comment:
remove the async since this function isn't async, and return `return Task.FromResult(metrics);`